### PR TITLE
Task definition and service should update at deploy time to match config + awesome specs

### PR DESCRIPTION
**Pull request by apurvis**
As discussed on https://github.com/lumoslabs/broadside/issues/31

also added a few things:

1. basic logging info when `bootstrap` doesn't do anything
1. refactored `update_task_revision` and `create_task_revision` to use the same `EcsManager` method

i tested the commands and while there's an issue with running the full deploy, i think it's some issue with beacon's mysql server.  @matl33t i'll message you outside of this PR.

@matl33t @rfroetscher

### CODE REVIEW

File: lib/broadside/deploy/ecs_deploy.rb [hunk 1]
```diff
@@ -158,32 +156,39 @@ def get_running_instance_ip
...
+    # Currently can only handle a single container definition.
     def update_task_revision
-      revision = create_new_task_revision
-
-      revision[:container_definitions].select { |c| c[:name] == family }.first.tap do |container_def|
-        container_def[:environment] = @deploy_config.env_vars
-        container_def[:image] = image_tag
-        container_def[:command] = @deploy_config.command
+      revision = EcsManager.get_latest_task_definition(family)
+      if revision[:container_definitions].size != 1
```
Comment (rfroetscher) [inline lib/broadside/deploy/ecs_deploy.rb]: why can't you look for the container definition that matches the family anymore?

File: lib/broadside/deploy/ecs_manager.rb [hunk 2]
```diff
@@ -11,6 +11,8 @@ class EcsManager
     }
 
     class << self
+      include Utils
```
Comment (rfroetscher) [inline lib/broadside/deploy/ecs_manager.rb]: why is this needed?

Comment (apurvis) [inline lib/broadside/deploy/ecs_manager.rb]: to enable the `debug`, `info`, `warn`, etc methods.

**Comment by apurvis** [comment]
> this seems to change the behavior of update_task_revision beyond just updating from the latest config because it no longer looks for the container def matching the family. why?

good question; there is no "why" and i will put that behavior back.

however i think most of the time with broadside - maybe _all_ the time - there is only one container definition in the task definition.

**Comment by rfroetscher** [comment]
not true. the airflow stuff has multiple

**Comment by apurvis** [comment]
huh, really?  
did you noticed that the current broadside code only updates the first of these container definitions - any changes to anything besides the first one will not propagate.

**Comment by rfroetscher** [comment]
it should only update the one with the same family. i have an nginx container that should never be touched by broadside

**Comment by apurvis** [comment]
cool i changed it, but just notice that it only ever updated the _first_ container definition in that family.

**Comment by apurvis** [comment]
there's definitely an issue here with changing existing task definitions in the case @rfroetscher mentioned which requires some working around... working on that now.

**Comment by apurvis** [comment]
k this should be ready for another look.  we still only handle the multiple configured container definitions as well as we always did except now we throw an exception when the code is not up to the demands of the config - specifically when there's > 1 `container_definition` of the same family in AWS already.  i don't think that's an issue we have and i didn't want to overengineer this.

**Comment by apurvis** [comment]
thanks to some advice from [an AWS dev](https://github.com/aws/aws-sdk-ruby/issues/1307#issuecomment-254368674), the broadside specs are now awesome.  you can peek into the entire call history by inspecting `api_request_log` and validate that the correct requests went out in the correct order.

File: lib/broadside/deploy/ecs_deploy.rb [hunk 3]
```diff
@@ -209,7 +223,7 @@ def run_command(command)
       command_name = command.join(' ')
       run_task_response = EcsManager.run_task(config.ecs.cluster, family, command)
 
-      unless run_task_response.successful?
+      unless run_task_response.successful? && run_task_response.tasks.try(:[], 0)
```
Comment (matl33t) [inline lib/broadside/deploy/ecs_deploy.rb]: what's the .try for?

**Comment by apurvis** [comment]
> what's the .try for?

to make sure it's an array and has something in it.  i got an error beefore because `run_task_response.tasks[0]` was `nil` and so the line below raised a "nil doesn't have `task_definition_arn` method"
